### ISSUE TITLE
SetSubnets not allowed on LB type network

### ISSUE THREAD

**Issue by DaveBlooman** [terraform-providers/terraform-provider-aws#1925]
Network load balancers cannot update subnets, so the following will return an error.

https://github.com/terraform-providers/terraform-provider-aws/blob/5eb6320f9f4628a255b871c234bee52c4cf83197/aws/resource_aws_lb.go#L400-L404

The example use case would be adding or removing a subnet from the list provided

I guess destroy then recreate could be offered, or just return an error indicating action isn't possible.

**Comment by bendk** [comment]
This is causing issues for me when I try to create a NLB

If I create without the subnets attribute, I get:
```
Error creating Application Load Balancer: ValidationError: At least one subnet must be specified
```

If I create with the subnets attribute, it creates the load balancer, but I see this error:
```
Failure Setting ALB Subnets: InvalidConfigurationRequest: SetSubnets is not supported for load balancers of type 'network' 
```

The second case isn't so bad, since the LB is created, and it doesn't fail when re-running terraform.  But it's anonying to say the least.

**Comment by catsby** [comment]
Fixed in #2310 

- creating a `aws_lb` with subnets was failing because the `UPDATE` method tried to re-apply and update the subnets, which isn't allowed. In #2310 we do not do the `UPDATE` of subnets if we're just now creating the load balancer
- updating subnets on network load balancer isn't allowed at all. In #2310 we can now mark `subnets` as a `ForceNew` operation if they are changed, so we won't try to update at all, and `terraform plan` will show the recreation

**Comment by catsby** [comment]
Reopening until #2310 is actually merged
